### ISSUE TITLE
Change interpretation of include_days in enw_filter_reference_dates

### ISSUE THREAD

**Issue by adrian-lison** [epinowcast/epinowcast#352]
The current implementation of `include_days` in `enw_filter_reference_dates` is not really intuitive to me, because it actually returns `include_days+1` reference dates.

 If I supply `include_days=10`, I expect to get exactly 10 reference dates back. If I supply `include_days=1`, I expect to get only the most recent reference date back. If I supply `include_days=0`, I expect to get no data back.
 
I think we should change this, knowing that it can be a breaking change for some running productive pieplines, and a lot of tests need to be updated.

**Comment by seabbs** [comment]
Yes, good catch I agree. I think as long as we provide a warning (potentially using `rlang` so it only shows up every 8 hours) and flag in the news that will be okay.
